### ISSUE TITLE
x.y operates on branded records, while x?.y operates on records

### ISSUE THREAD

**Issue by kach** [accordproject/ergo#385]
```
ergo$ define concept X {a: Integer}
```
So `.` only works on branded records, which seems reasonable.
```
ergo$ return X{a:1}.a;
Response. 1
  :  {emit: Unit[], response: Integer, state: Unit}?
ergo$ return {a:1}.a;
TypeError (at line 1 col 7). This operator received an unexpected argument of type `{a: Integer}'
```
But this is inconsistent with `?.`, which only works on _unbranded_ records.
```
ergo$ return some(X{a: 1})?.a;
TypeError (at line 1 col 7). This operator received an unexpected argument of type `~org.accordproject.ergotop.X'
return some(X{a: 1})?.a;
       ^^^^^^^^^^^^^^^^ 
ergo$ return some({a: 1})?.a;
Response. 1
  :  {emit: Unit[], response: Any, state: Unit}?
```

**Comment by jeromesimeon** [comment]
Ouch. Thanks for the report. That's definitely a bug -- we should also have an operator that performs `.` on records. Possibly see #375

**Comment by jeromesimeon** [comment]
Lots of problems to disentangle here. One if that the expansion of `?.a` into a match was completely wrong in the first place. Two distinct problems:
1. It should apply to objects rather than records -- as pointed out -- in order to be consistent with `.a`
2. The case where `some x` is matched should return a `some y` back

An on-going fix for those already looks a little better:
```
$ ./bin/ergotop
Welcome to ERGOTOP version 0.3.4
ergo$ define concept B{ b : Integer }
ergo$ define constant opt1 : B? = some(B{ b : 1})
ergo$ define constant opt2 : B? = none
ergo$ return opt1?.b;
Response. some(1) : Integer?
ergo$ return opt2?.b;
Response. none : Integer?
ergo$ 
```

**Comment by jeromesimeon** [comment]
Fixed in #499
